### ISSUE TITLE
Exception Value: can't pickle module objects

### ISSUE THREAD

**Issue by kevinrenskers** [flashingpumpkin/django-socialregistration#73]
When I try to login with any OpenID provider, I get the error "can't pickle module objects" on the callback. Probably because I use django.contrib.sessions.backends.cache as my session engine. When debugging, I found that the memcache module tries to pickle this dictionary:

```
{'next': '/profiel/',
 'openid_provider': u'http://yahoo.com/',
 'socialregistration_client': <socialregistration.utils.OpenID object at 0x9cc344c>,
 'socialregistration_profile': <OpenIDProfile: OpenID profile for None, via provider None>,
 'socialregistration_user': <User: >}
```

It looks like your "del request.session['socialregistration_user']" fix in views.py doens't work?

Exception Type: TypeError at /social/openid/callback/
Exception Value: can't pickle module objects
Complete error is pasted to http://pastie.org/1807108

**Comment by flashingpumpkin** [comment]
What commit are you running off?

It's not actually the `request.session['socialregistration_user']` that is causing this error to happen, but `request.session['socialregistration_client']` - which in all cases except for the Facebook client holds the request, which in turn holds the session, which in turn holds ... you get the point.

The following line should make sure that this does not happen:
https://github.com/flashingpumpkin/django-socialregistration/blob/master/socialregistration/views.py#L319

I've just tested connecting via OpenID from my Google account and it seems to work fine. 

In case you do a pull and the issue still happens, I've got a feeling that there's still some bits missing from the traceback above. Make sure you post the whole thing.

**Comment by flashingpumpkin** [comment]
Ah. Yep. It's an issue when using the cache backend for sessions. Looking into it

**Comment by kevinrenskers** [comment]
Yeah, no problems when using DB backend for sessions.

**Comment by flashingpumpkin** [comment]
Yeah, I think I can track it back to the point where `openid.consumer.consumer.Consumer` expects some sort of dict as datastore, and I'm giving it the `request.session` object. Unsetting `client.consumer` fixes the problem, but obviously renders it unusable when getting it back out of the session. I'll have a look later how it behaves when only using a dict instead of the session object as storage. If it's still throwing an error then, the problem lies within `openid.Consumer` and how it is importing modules. 

We shall see.

**Comment by flashingpumpkin** [comment]
From the looks of it there needs some refactoring to be done to make everything pickle properly with memcache.

**Comment by flashingpumpkin** [comment]
Fixed with 0.5 - no more objects / sockets going into the session that can't be pickled.
